specification/azsadmin/resource-manager/storage/Microsoft.Storage.Admin/preview/2016-05-01/farms.json

```diff
@@ -0,0 +1,709 @@
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK -- The new storage farm has been created.",
+                        "schema": {
+                            "$ref": "#/definitions/Farm"
+                        }
+                    }
+                }
+            },
+            "get": {
+                "x-ms-examples": {
+                    "Returns the Storage properties and settings for a specified storage farm.": {
+                        "$ref": "./Examples/Farms/Get.json"
+                    }
+                },
+                "tags": [
+                    "Farms"
+                ],
+                "description": "Returns the Storage properties and settings for a specified storage farm.",
+                "operationId": "Farms_Get",
+                "parameters": [
+                    {
+                        "$ref": "storage.json#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/ResourceGroupNameParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/FarmIdParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/ApiVersionParameter"
+                    }
+                ],
+                "responses": {
+                    "200": {
+                        "description": "OK -- The farm has been returned.",
+                        "schema": {
+                            "$ref": "#/definitions/Farm"
+                        }
+                    },
+                    "404": {
+                        "description": "NOT FOUND -- The specified farm was not found."
+                    }
+                }
+            },
+            "patch": {
+                "x-ms-examples": {
+                    "Update an existing storage farm.": {
+                        "$ref": "./Examples/Farms/Update.json"
+                    }
+                },
+                "tags": [
+                    "Farms"
+                ],
+                "description": "Update an existing storage farm.",
+                "operationId": "Farms_Update",
+                "parameters": [
+                    {
+                        "$ref": "storage.json#/parameters/SubscriptionIdParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/ApiVersionParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/ResourceGroupNameParameter"
+                    },
+                    {
+                        "$ref": "storage.json#/parameters/FarmIdParameter"
+                    },
+                    {
+                        "$ref": "#/parameters/FarmObject"

```

Comment:
Please directly define body param here like:

```
          {
            "name": "farmObject",
            "in": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/Farm"
            },
            "description": "Farm object supplied in the body of the Patch farm operation."
          }
```

and remove `FarmObject` from parameter section (/parameters/FarmObject)

Reply:
Done, but not sure why this was needed.  Could you explain?

Reply:
There was linter validation error in this spec, when declaring any parameter in parameter section we need to explicitly indicate that where exactly parameter needs to be declared (as client property, as method parameter). This specific parameter was missing that indication. But noted that this param is used only here, so I thought let's move it local to this operation so it is more readable & also we don't have to explicitly indicate it's decl location when specifying locally.